integrations/jenkins/client.py

```diff
@@ -75,8 +75,9 @@ async def fetch_resources(
             if len(jobs) < page_size:
                 break
 
-        # now process child jobs
-        yield await self._process_jobs(resource, child_jobs)
+        for job in child_jobs:
```

Comment:
immediately yield every job found rather than collecting them throughout all the levels